staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_discovery_test.go

```diff
@@ -240,6 +250,58 @@ func checkAPIGroups(t *testing.T, api apidiscoveryv2beta1.APIGroupDiscoveryList,
 	}
 }
 
+// TestInitialRunHasAllAPIServices tests that when discovery is ready, all APIService
+// are present and ones that have not synced are in the list as Stale.
+func TestInitialRunHasAllAPIServices(t *testing.T) {
```

Comment:
I think this sort of thing needs an integration test instead. Make an aggregated APIservice that takes a long time to respond. The healthz shouldn't be up until you let your fake apiservice respond.

I have a utlity in here somewhere that makes it easy to set up these sorts of integration tests:

https://github.com/kubernetes/kubernetes/blob/233949e05df3020c4ca58448343352d87f7a4493/test/integration/apiserver/discovery/discovery_test.go#L175-L212